File: samples/sample-app-storage/src/main/java/com/azure/android/storage/sample/config/StorageConfiguration.java
package com.azure.android.storage.sample.config;

import android.content.Context;

import com.azure.android.storage.sample.R;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.annotations.SerializedName;

import java.io.IOException;
import java.io.InputStream;

public class StorageConfiguration {
    @SerializedName("blob_service_url")
    String mBlobServiceUrl;

    @SerializedName("container_name")
    String mContainerName;

    public String getBlobServiceUrl() {
        return mBlobServiceUrl;
    }

    public String getContainerName() {
        return mContainerName;
    }

    public static StorageConfiguration create(Context context) {
        byte[] buffer;
        try(InputStream configStream = context.getResources().openRawResource(R.raw.storage_configuration)) {
            buffer = new byte[configStream.available()];
            configStream.read(buffer);
        } catch (IOException e) {
            throw new IllegalArgumentException("Unable to open the configuration file - 'R.raw.storage_configuration'.", e);
        }
        final String config = new String(buffer);
        final Gson gson = new GsonBuilder().create();
        return gson.fromJson(config, StorageConfiguration.class);
    }
}

File: samples/sample-app-storage-kotlin/src/main/java/com/azure/android/storage/sample/kotlin/config/StorageConfiguration.kt
package com.azure.android.storage.sample.kotlin.config

import android.content.Context
import com.azure.android.storage.sample.kotlin.R
import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName
import java.io.IOException

class StorageConfiguration {
    @SerializedName("blob_service_url")
    var blobServiceUrl: String? = null

    @SerializedName("container_name")
    var containerName: String? = null

    @SerializedName("sas_token")
    var sasToken: String? = null

    companion object {
        fun create(context: Context): StorageConfiguration {
            var buffer: ByteArray
            try {
                context.resources.openRawResource(R.raw.storage_configuration).use { configStream ->
                    buffer = ByteArray(configStream.available())
                    configStream.read(buffer)
                    val configBytes = String(buffer)
                    val gson = GsonBuilder().create()
                    val config = gson.fromJson(configBytes, StorageConfiguration::class.java)
                    if (!config.blobServiceUrl?.endsWith("/")!!) {
                        config.blobServiceUrl += "/"
                    }
                    return config;
                }
            } catch (e: IOException) {
                throw IllegalArgumentException("Unable to open the configuration file - 'R.raw.storage_configuration'.", e)
            }
        }
    }
}
